ci-operator/step-registry/multiarch/validate-nodes/multiarch-validate-nodes-commands.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#######################################################################################################################
+# The following logic will be better refactored as part of https://issues.redhat.com/browse/MULTIARCH-4846.
+# - On AWS, when CONTROL_ARCH="amd64" (def) and COMPUTE_ARCH="amd64" (def) and OCP_ARCH="arm64", OCP_ARCH prevails.
+# - On GCP, when CONTROL_ARCH="" (def) and COMPUTE_ARCH="" (def), they are set to the OCP_ARCH value.
+# - Other cloud providers do not implement the CONTROL_ARCH or COMPUTE_ARCH variables at time of writing.
+#   Therefore, this step will be skipped for them.
+# At steady state, we would like this step to be onboarded to all cloud providers,
+# and the logic to be coherent across all of them.
+if [ -z "${CONTROL_ARCH}" ] || [ -z "${COMPUTE_ARCH}" ]; then
+  echo "[INFO] CONTROL_ARCH and COMPUTE_ARCH are not set. Skipping this step."
+  exit 0
+fi
+CONTROL_ARCH="${CONTROL_ARCH:-${OCP_ARCH}}"
+COMPUTE_ARCH="${COMPUTE_ARCH:-${OCP_ARCH}}"
+# In the AWS case, if OCP_ARCH is set to arm64, but CONTROL_ARCH and COMPUTE_ARCH are left to the default values,
+# we need to adjust them to arm64.
+# We can state that this is valid for all the cloud providers, because no users should set both
+# OCP_ARCH, CONTROL_ARCH and COMPUTE_ARCH in their job configuration.
+# See https://github.com/openshift/release/blob/c6ab6d5cdc70/ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh#L88-L91
+# for AWS.
+if [ "${OCP_ARCH}" == "arm64" ]; then
```

Comment:
We do not need this `if` block any more when we already have the line 19 and 20, right ?

Reply:
I just tried to explain the missing coherence in the comments. Unless I'm missing something, I thought we still needed it, don't we?

Reply:
> AWS with OCP ARCH set to arm64

When OCP_ARCH set to arm64, the line 19 and 20 will set `CONTROL_ARCH` and `COMPUTE_ARCH` to `arm64`, right?

Reply:
You're absolutely right, and at this point, it seems a lot safer to apply it as a verification for any clusters. I updated the code to reflect the same strategy as the node readiness check: install-config, if any, then (CONTROL|COMPUTE)_ARCH, then OCP_ARCH (default to amd64).